socketDataHandler.js

```diff
@@ -46,25 +51,80 @@ class SocketDataHandler {
       this.keepAliveTimeout.push(setTimeout(() => this.socket.destroy(), keepAliveMaxWaitForResponse))
       this.socket.write(this.keepAlivePacket.loadIntoBuffer());
     }
-  
+
+    handshake(packet) {
```

Comment:
Moved logic from the monolithic socketData method into smaller methods for clarity